nexus/types/src/lib.rs

```diff
@@ -0,0 +1,7 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
I think it'd help to have some module-level docs for nexus::types to explain what goes here and why.